pkg/didcomm/protocol/introduce/service.go

```diff
@@ -161,48 +126,25 @@ func New(p Provider) (*Service, error) {
 		didEventService: didService,
 		callbacks:       make(chan *metaData),
 		didEvent:        make(chan service.StateMsg),
-		stop:            make(chan struct{}),
 	}
 
 	if err = svc.didEventService.RegisterMsgEvent(svc.didEvent); err != nil {
 		return nil, fmt.Errorf("did register msg event: %w", err)
 	}
 
 	// start the listener
-	svc.wg.Add(1)
-
 	go svc.startInternalListener()
 
 	return svc, nil
 }
 
-// Stop stops service (callback listener)
-func (s *Service) Stop() error {
-	s.closedMutex.Lock()
-	defer s.closedMutex.Unlock()
-
-	if s.closed {
-		return ErrServerWasStopped
-	}
-
-	if err := s.didEventService.UnregisterMsgEvent(s.didEvent); err != nil {
-		return fmt.Errorf("stop unregister msg event: %w", err)
-	}
-
-	close(s.stop)
-	s.closed = true
-	s.wg.Wait()
-
-	return nil
-}
-
 // startInternalListener listens to messages in gochannel for callback messages from clients.
 func (s *Service) startInternalListener() {
 	for {
 		select {
 		case msg := <-s.callbacks:
-			// if no error - do handle or it was disapproved
-			if msg.err == nil || msg.disapprove {
+			// if no error or it was rejected do handle
+			if msg.err == nil || msg.rejected {
```

Comment:
Should the service continue to handle the message in case of rejected ? ie `rejected=true`. I would have imagined the service to handle message when the consumer sets rejected to false or not set it at all (default).

Reply:
`rejected` is an internal field. User has two possible options:
* Continue(...) 
* Stop()
When user executes `Stop` function protocol sets `rejected=true`. Then we send a `response` message with approve false.